core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -537,7 +538,8 @@ class ReplicaManager(val config: KafkaConfig,
           val initialLogEndOffset = localReplica.logEndOffset
           val logReadInfo = localReplica.log match {
             case Some(log) =>
-              log.read(offset, fetchSize, maxOffsetOpt)
+              val adjustedFetchSize = if (TopicConstants.INTERNAL_TOPICS.contains(topic) && !readOnlyCommitted) Math.max(fetchSize, log.config.maxMessageSize) else fetchSize
```

Comment:
@becketqin Makes sense -- I guess aside from fetches we don't actually limit the size of responses anyway, so this isn't much different from handling other types of requests. In fact, in reviewing this I noticed that although we have a bunch of logic to put constraints on the receive size, we never use any value other than `UNLIMITED` when creating `Selectors`....